caffe2/core/operator_schema.h

```diff
@@ -478,9 +478,9 @@ inline vector<TIndex> GetDimsVector(const TensorShape& shape) {
 }
 
 // Helper function
-inline uint64_t nElemFromDim(const TensorShape& X) {
-  uint64_t nElem = X.dims_size() > 0 ? 1 : 0;
-  for (int i = 0; i < X.dims_size(); ++i) {
+inline uint64_t nElemFromDim(const TensorShape& X, int dim = 0) {
+  uint64_t nElem = X.dims_size() > dim ? 1 : 0;
+  for (int i = dim; i < X.dims_size(); ++i) {
```

Comment:
Could we use `size_from_dim_` here?

Reply:
One issue is TensorShape is not a vector and operator_schema.h doesn't include tensor.h as far as I can tell.

Reply:
btw, 0! == 1. Don't we want to report 1 for overshooting dims? Or may be even throw in this case? 0 is kinda surprising.

Reply:
Since this is a utility function we may want to be conservative about throwing an error? For example, what if by some reason one uses an empty FC/Conv bias as an empty tensor? So maybe a warning?

Reply:
okay, size_from_dim in tensor.h returns 1 for such case. We probably should be consistent. This is also consistent with 0!